### ISSUE TITLE
[Pioneer] - Android 8.1 - Device needs significant amount of time to acquire GSM connectivity

### ISSUE THREAD

**Issue by dimitar10** [sonyxperiadev/bug_tracker#121]
**Platform:** Nile
**Device:** Pioneer H4113
**Kernel version:** 4.4.124 - (prebuilt @ ffba6bd -dirty, branch: aosp/LA.UM.6.4.r1)
**Android version:** 8.1.0r20
**Firmware version:** 8.1.6.4_r1_v11_nile

**Description** After boot up, the device always need **about 2min 30 sec** to acquire GSM connectivity, in which time the device behaves like it's in airplane mode. This was tested with several LTE-capable SIM cards, with and without a pre-set pin.

**Symptoms** The GSM signal meter in the notification bar shows that there is no connectivity. Additionally, if the user tries to make a call, the ``Turn off airplane mode to make a call`` message is displayed, and a quick ``Radio off`` message is displayed in the in-call screen.

**How to reproduce**
After a fresh flash or a factory data reset, wait for the boot process to finish and on the home screen the following message will appear:

``Update preferred SIM card? <Provider> is the only SIM in your device. Do you want to use this SIM for mobile data, calls, and SMS messages?``

After selecting, for instance, ``YES``, the following message will appear:

``Switching data SIM, this may take up to a minute...``

After the above mentioned time is elapsed, the phone will acquire GSM connectivity. After every subsequent reboot, the device will require about the same time to acquire connectivity.

**Logs**
Attached to this message you can find the logcat and dmesg.

[gsm-connectivity-delayed.tar.gz](https://github.com/sonyxperiadev/bug_tracker/files/1942790/gsm-connectivity-delayed.tar.gz)

**Comment by jerpelea** [comment]
this is not a bug
device is slower

**Comment by dimitar10** [comment]
Thank you for the input, Alin. I will close the issue...

**Comment by dimitar10** [comment]
Platform: Nile
Device: Pioneer H4113
Kernel version: 4.4.139-g880676b
Android version: 8.1.0r35
Firmware version: 8.1.6.4_r1_v13_nile

**Description**

GSM connectivity cannot be established in the first couple of minutes after reboot (~2:20) or after disabling Airplane mode.

**How to reproduce**

    Reboot device
    Try to initiate a GSM call

**OR**

    Turn Airplane mode ON
    Turn Airplane mode OFF
    Try to initiate a GSM call

**Outcome**: Mobile network not available

**Note**: The period of initial GSM inactivity can be reduced by ~30 seconds, by forcing a network search, after disabling Airplane Mode, but even this option is dysfunctional in the first ~70 seconds following the Airplane Mode disabling. Around the 1 min 10 sec mark, the network search can be successfully started, leading to a network connection around 1 min 45 seconds.

**Discussion**: Re-tested this issue, and compared it to a stock Pioneer device, and the stock device is significantly quicker in establishing GSM connectivity (almost instantly after disabling Airplane mode and before the home screen is shown after boot up). 

Is there any possible way to mitigate this issue i.e. to at least reduce the time for acquiring GSM connectivity?

**Logs**: dmesg and logcat included

[gsm-issue-20180725.tar.gz](https://github.com/sonyxperiadev/bug_tracker/files/2227061/gsm-issue-20180725.tar.gz)

**Comment by dimitar10** [comment]
Re-tested with:

**Platform:** nile
**Device:** pioneer (H4113)
**Kernel version:** 4.4.140-gdc5bc38
**Android version:** 8.1.0-r35
**Firmware version:** 8.1.6.4_r1_v14_nile

Outcome is the same.

**Comment by mirh** [comment]
Could this be related to #141?

**Comment by jerpelea** [comment]
please set the following paramter in the build 
ro.telephony.default_network=9,0

thanks

**Comment by jerpelea** [comment]
@dimitar10 please test this commit
https://github.com/sonyxperiadev/device-sony-pioneer/commit/8e138d84b1c0c7763929ba910c592dd7bd0055eb

**Comment by dimitar10** [comment]
Re-tested again, but the results are the same. First, I flashed with stock Sony firmware @ 50.1.A.10.40, booted up and then flashed AOSP 8.1.0_r35 with kernel 4.4.144-g2c9bf97 and v14 firmware.

Because the issue is not present on the H3113 (single-sim) device, I tried disabling the dual-sim capability of the H4113 by editing `aosp_h4113.mk`, firstly by removing and another attempt at editing the values for the multisim properties from `dsds` to `none`  but both cases went horribly wrong i.e. the device will acquire GSM connectivity quickly only if the device had acquired connectivity "beforehand" and you never remove/reinsert the SIM card. After the first removal of the SIM card, the device will never obtain GSM connectivity again.

Under `beforehand` I mean from a previous flash, but it should be noted that I flashed all of the .img files, including the v14 firmware.

As a side-question, is there a correct way to disable the dual-sim capability on the H4113?

**Comment by jerpelea** [comment]
I tested the same with my sim and the network si aquired in a normal time. I wonder if this is carrier dependent,

**Comment by jerpelea** [comment]
The investigation is ongoing 

In the meantime for the dual sim devices that use only 1 sim there is a workaround 
setprop persist.radio.multisim.config none
setprop persist.multisim.config none
reboot

**Comment by jerpelea** [comment]
please test the commit
sonyxperiadev/device-sony-common@9d8c552

**Comment by jerpelea** [comment]
@dimitar10 can you please test the fix

**Comment by dimitar10** [comment]
@jerpelea Sorry I didn't update the issue earlier... The commit fixes the issue. Tested both for a single SIM card for slots 1 and 2 when the device is acquiring GSM connectivity after reboot, and with airplane mode switching on/off.

Additionally I tested with two SIM cards, and everything appears to be functional as well. 

One little thing that I noticed is that the SIM pin lock screen seems to be dismissed too early in some cases (e.g. first SIM is unlocked, second SIM is locked with a pin etc.), but anyways, if this proves to be more than just a nuisance I will report it as a separate issue.
